okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java

```diff
@@ -2080,6 +2080,23 @@ public void shutdownNow_streamListenerRpcProgress() throws Exception {
     assertEquals(MISCARRIED, listener3.rpcProgress);
   }
 
+  @Test
+  public void finishedStreamRemovedFromInUseState() throws Exception {
+    initTransport();
+    setMaxConcurrentStreams(1);
+    final MockStreamListener listener = new MockStreamListener();
+    OkHttpClientStream stream =
+        clientTransport.newStream(method, new Metadata(), CallOptions.DEFAULT, tracers);
+    stream.start(listener);
+    OkHttpClientStream pendingStream =
+        clientTransport.newStream(method, new Metadata(), CallOptions.DEFAULT, tracers);
+    pendingStream.start(listener);
+    waitForStreamPending(1);
+    clientTransport.finishStream(stream.transportState().id(), Status.OK, PROCESSED,
+        false, null, null);
+    assertFalse(clientTransport.inUseState().anyObjectInUse(stream));
```

Comment:
Instead of this white-box where you dig into the internals, we can easily test that the results are correct instead. This bug causes `transportListener.transportInUse()` not to be updated appropriately. You can do `verify(transportListener).transportInUse(true/false)` instead.